tests/clients/test_environment.py

```diff
@@ -20,33 +20,302 @@
+ENVIRONMENT_CREATION_RESPONSE = EnvironmentCreationResponse(id=ENVIRONMENT_ID)
+
+ENVIRONMENT_CREATE_UPDATE_BODY = {
+    "name": NAME,
+    "description": DESCRIPTION,
+    "specification": SPECIFICATION_BODY,
+}
+ENVIRONMENT_CREATE_UPDATE = EnvironmentCreateUpdate(
+    name=NAME, description=DESCRIPTION, specification=SPECIFICATION
+)
+ENVIRONMENT_CREATE_UPDATE_BODY_NO_DESCRIPTION = {
+    "name": NAME,
+    "description": None,
+    "specification": SPECIFICATION_BODY,
 }
+ENVIRONMENT_CREATE_UPDATE_NO_DESCRIPTION = EnvironmentCreateUpdate(
+    name=NAME, description=None, specification=SPECIFICATION
+)
+
+
+def test_version_schema_load():
+    data = VersionSchema().load(VERSION_BODY)
+    assert data == VERSION
+
+
+def test_version_schema_dump():
+    data = VersionSchema().dump(VERSION)
+    assert data == VERSION_BODY
+
+
+def test_version_schema_load_invalid():
+    with pytest.raises(ValidationError):
+        VersionSchema().load(INVALID_VERSION_BODY)
+
+
+def test_version_schema_dump_invalid():
+    with pytest.raises(ValidationError):
+        VersionSchema().dump(INVALID_VERSION)
+
+
+@pytest.mark.parametrize(
+    "body, expected",
+    [
+        (PYTHON_PACKAGE_BODY, PYTHON_PACKAGE),
+        (PYTHON_PACKAGE_BODY_LATEST, PYTHON_PACKAGE_LATEST),
+    ],
+)
+def test_python_package_schema_load(body, expected):
+    data = PythonPackageSchema().load(body)
+    assert data == expected
+
+
+@pytest.mark.parametrize(
+    "object, expected",
+    [
+        (PYTHON_PACKAGE, PYTHON_PACKAGE_BODY),
+        (PYTHON_PACKAGE_LATEST, PYTHON_PACKAGE_BODY_LATEST),
+    ],
+)
+def test_python_package_schema_dump(object, expected):
+    data = PythonPackageSchema().dump(object)
+    assert data == expected
+
+
+def test_pip_schema_load():
+    data = PipSchema().load(PIP_BODY)
+    assert data == PIP
+
+
+def test_pip_schema_dump():
+    data = PipSchema().dump(PIP)
+    assert data == PIP_BODY
+
+
+def test_conda_schema_load():
+    data = CondaSchema().load(CONDA_BODY)
+    assert data == CONDA
+
+
+def test_conda_schema_dump():
+    data = CondaSchema().dump(CONDA)
+    assert data == CONDA_BODY
+
+
+def test_python_specification_schema_load():
+    data = PythonSpecificationSchema().load(PYTHON_SPECIFICATION_BODY)
+    assert data == PYTHON_SPECIFICATION
+
+
+def test_python_specification_schema_dump():
+    data = PythonSpecificationSchema().dump(PYTHON_SPECIFICATION)
+    assert data == PYTHON_SPECIFICATION_BODY
+
+
+def test_apt_package_schema_load():
+    data = AptPackageSchema().load(APT_PACKAGE_BODY)
+    assert data == APT_PACKAGE
+
+
+def test_apt_package_schema_dump():
+    data = AptPackageSchema().dump(APT_PACKAGE)
+    assert data == APT_PACKAGE_BODY
+
+
+def test_apt_schema_load():
+    data = AptSchema().load(APT_BODY)
+    assert data == APT
+
+
+def test_apt_schema_dump():
+    data = AptSchema().dump(APT)
+    assert data == APT_BODY
+
+
+def test_python_schema_load():
```

Comment:
Can we also test load and dump in the case that Python2 and/or Python3 are unset (None / not present in the BODY dict)? I'm not sure that would work at the moment (I think the fields in the `PythonSchema` may require a `missing=None` argument each).